### ISSUE TITLE
Transculent particles flickering

### ISSUE THREAD

**Issue by Azmalent** [CyclopsMC/EverlastingAbilities#160]
#### Issue type:

- :bug: Bug
____

#### Short description:

When Everlasting Abilities is installed, transparent particles will randomly turn opaque.
Here is an example with firefly jar from Twilight Forest (TF is only used for demonstration purposes, it is unrelated to the issue as particles from other mods are also affected)
![Gif link](https://s5.gifyu.com/images/firefly.gif)

This was tested with no other mods than Cyclops Core, Everlasting Abilities and Twilight Forest.

#### Steps to reproduce the problem:

1. Install Everlating Abilities
2. Look at transparent particles

#### Expected behaviour:

Particles should render without flickering.

____

#### Versions:

- Cyclops Core: 1.6.1
- Everlasting Abilities: 1.5.2
- Minecraft: 1.12.2
- Forge: 14.23.5.2847

**Comment by rubensworks** [comment]
Thanks for reporting!

**Comment by rubensworks** [comment]
I just looked into it, and I'm pretty sure that this is a problem with TF.

If I remove [this line](https://github.com/CyclopsMC/CyclopsCore/blob/master-1.12/src/main/java/org/cyclops/cyclopscore/client/particle/ParticleBlur.java#L132), the TF particles show up normally. However, other particles start rendering strangely, such as the breaking block particles.

@Azmalent Could you report this at the TF issue tracker? I suspect they are missing a `GlStateManager.enableBlend()` line before their rendering logic.
Feel free to mention me in the issue, happy to help where needed.

**Comment by Tamaized** [comment]
and are we certain that the blend state is ever disabled before the render... this sounds like a state leak on your end unless im mistaken

**Comment by rubensworks** [comment]
@Tamaized 

> this sounds like a state leak on your end unless im mistaken

That was my initial thought as well.
However, since after removing `GlStateManager.disableBlend()`, vanilla particles (such as block breaking) start having a transparency, which seems to indicate that vanilla expects blending to be disabled by default.

**Comment by SonicX8000** [comment]
I noticed something with particles from the Betweenland's Item Magnet which pulls items towards you. When the items are being attracted towards you they give off these tiny light blue sparks that are suppose to fade away after awhile but... if Everlasting Abilities is installed... two of these can occur with those particles...

A. The sparks no longer fade away when they're disappearing. It goes from opaque to nothing when it actually disappears.

B. No sparks render at all.

C. The sparks fade away but then flash opaque for a split second much like how the particles from the firefly jar in OP's image.

-----

There also this thing I've noticed while playing in a private pack but due to the many mods installed in there... this may not even be helpful but still figure I give it a mention.

Living Enchantment's Purple/Light Blue XP Orbs become transparent when you look at them at certain angles. They're suppose to be opaque. It's possible for these Orbs to become transparent while the other ones are opaque.

=====

Now... before I installed Everlasting Abilities, none of the above issues occurred and only started to happen when I installed Everlasting Abilities. This is within Minecraft 1.12.2.
